### PR TITLE
Removing multiple unlike from one user on a post

### LINKED ISSUE #23: Multiple unlike from one user on a post
**Describe the bug**
 when a user do unlike  multiple times on a same post  then like count is decremented  and goes to negative number field. 

**To Reproduce**
1. Go to 'http://localhost:8000/api-docs/'
2. Scroll down to ```post``` docs
3. Click on  ```/api/v1/home/unlike-post``` .Fill all the necessary fields to run the end point.
4. Click  on execute button more then one times .then you will find the like count is negative number and that is not good.

**Screenshots**
``` Json
   "likeCount": -2,
    "commentCount": 0,
    "likes": [],
    "comments": [],
    "genreName": [],
    "genreId": [
        2,
        5,
        6
    ],
```

**Desktop (please complete the following information):**
 - OS: ``` Windows```
 - Browser : ```Chrome```

### PR DESCRIPTION
**Pull request by GURUDATTDAHARE**
### Description
I remove multiple unlike from one user on a post means a user can unlike post only one time on a post. 

**Type of Change:** (Delete irrelevant options)
- Code

**Code/Quality Assurance Only** (Delete irrelevant options)

- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [ ] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: routes/postRoutes.js [hunk 1]
```diff
@@ -160,28 +160,33 @@ exports.likePost = (req, res) => {
...
-          $pull: { likes: req.user._id },
-          $inc: { likeCount: -1 },
-        },
-        { new: true }
-      ).exec((err, result) => {
-        if (err) return res.status(422).json({ error: err });
-        else return res.status(201).json(result);
-      });
+    .then((docs) => {
+      if (docs === null) {
```
Comment (KunalAnkur) [inline routes/postRoutes.js:172]: Can you show me what data contain in `docs`? In both cases.

**Review by KunalAnkur — COMMENTED** [review]
Looking good!! Look below, I have some queries which I left

File: routes/postRoutes.js (hunk 1 above)
Comment (GURUDATTDAHARE) [reply routes/postRoutes.js:172]: ## In the basis to previous code:
- first i create a post and add a like on this post, the response will be 
 ```Json
  "likeCount": 1,
    "commentCount": 0,
    "likes": [
        "6082b6c4f1442b308442ff3b"
    ],
    "comments": [],
    "genreName": [],
    "genreId": [
        2,
        5,
        6
    ],
```
- after that , Now i unlike the post at 3 times ,the response will be
```Json
"likeCount": -2,
    "commentCount": 0,
    "likes": [],
    "comments": [],
    "genreName": [],
    "genreId": [
        2,
        5,
        6
    ],
``` 
here the  person who liked this post his data was deleted from database and like count is decremented on negative field.

## In the basis of new code
 - first i create a post and add a like on this post, the response will be 
 ```Json
  "likeCount": 1,
    "commentCount": 0,
    "likes": [
        "6082b6c4f1442b308442ff3b"
    ],
    "comments": [],
    "genreName": [],
    "genreId": [
        2,
        5,
        6
    ],
```
- Now i unlike this post many times but the like count  is decrement  only  one time.
```Json
 "likeCount": 0,
    "commentCount": 0,
    "likes": [],
    "comments": [],
    "genreName": [],
    "genreId": [
        2,
        5,
        6
    ],
```

Comment (KunalAnkur) [reply routes/postRoutes.js:172]: That's so great but I am talking about `docs` variable. Can you show me what it contains?

**Pull request merged**